### PR TITLE
Enable unparam linter

### PR DESCRIPTION
**Pull request by twz123**
## Description

... and fix lints on the way.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: internal/pkg/file/atomic_test.go [hunk 1]
```diff
@@ -84,10 +84,12 @@ func TestWriteAtomically(t *testing.T) {
...
 	}
 
-	assertDirEmpty := func(t *testing.T, dir string) bool {
+	assertDirEmpty := func(t *testing.T, dir string) {
 		t.Helper()
 		entries, err := os.ReadDir(dir)
-		return assert.NoError(t, err) && assert.Empty(t, entries)
+		if assert.NoError(t, err) {
+			assert.Empty(t, entries)
+		}
```
Comment (kke) [inline internal/pkg/file/atomic_test.go:92]:
```suggestion
		assert.Empty(t, entries)
		assert.NoError(t, err) 
```

File: pkg/component/controller/workerconfig/reconciler_test.go [hunk 2]
```diff
@@ -374,22 +374,22 @@ func TestReconciler_ResourceGeneration(t *testing.T) {
...
 		},
 
-		"worker-config-profile_XXX-1.31": func(t *testing.T, expected *kubeletConfig) {
+		"worker-config-profile_XXX-1.31": func(_ *testing.T, expected *kubeletConfig) {
 			expected.Authentication.Anonymous.Enabled = ptr.To(true)
 			expected.FeatureGates = map[string]bool{"kubelet-feature": true}
 		},
 
-		"worker-config-profile_YYY-1.31": func(t *testing.T, expected *kubeletConfig) {
+		"worker-config-profile_YYY-1.31": func(_ *testing.T, expected *kubeletConfig) {
```
Comment (kke) [inline pkg/component/controller/workerconfig/reconciler_test.go:392]: None of  them use `testing.T`. Name `configMaps` could be improved.

File: pkg/kubernetes/watch/watcher_test.go [hunk 3]
```diff
@@ -497,7 +497,7 @@ func TestWatcher(t *testing.T) {
 
 			return nil
 		}
-		second = func(opts metav1.ListOptions) error {
+		second = func(opts metav1.ListOptions) (unused error) { // nolint:unparam
```
Comment (kke) [inline pkg/kubernetes/watch/watcher_test.go:500]: Maybe `(_ error)`?

File: pkg/component/controller/workerconfig/reconciler_test.go (hunk 2 above)
Comment (twz123) [reply pkg/component/controller/workerconfig/reconciler_test.go]: Done. PTAL.

File: pkg/kubernetes/watch/watcher_test.go (hunk 3 above)
Comment (twz123) [reply pkg/kubernetes/watch/watcher_test.go]: I've opted to just add a comment to the nolint directive.

**Review by kke — APPROVED** [review]

**Review by kke — APPROVED** [review]

**Pull request merged**